src/index.ts

```diff
@@ -30,11 +27,72 @@ const transact = (doc: Y.Doc | null, opts: Options, fn: () => void) => {
   }
 };
 
+const toYValue = (val: any) => {
+  if (isProxyArray(val)) {
+    const arr = new Y.Array();
+    arr.insert(
+      0,
+      val.map(toYValue).filter((v) => v !== undefined && v !== null),
+    );
+    return arr;
+  }
+
+  if (isProxyObject(val)) {
+    const map = new Y.Map();
+    Object.entries(val).forEach(([key, value]) => {
+      const v = toYValue(value);
+      if (v !== undefined) {
+        map.set(key, v);
+      }
+    });
+    return map;
+  }
+
+  if (isPrimitiveMapValue(val)) {
+    return val;
+  }
+
+  return undefined;
+};
+
+const toJSON = (yv: unknown) => {
+  if (yv instanceof Y.AbstractType) {
+    return yv.toJSON();
+  }
+
+  return yv;
+};
+
+const getNestedValues = <T>(
+  p: Record<string, T> | T[],
+  y: Y.Map<T> | Y.Array<T>,
+  path: (string | number)[],
+) => {
+  let pv: any = p;
+  let yv: any = y;
+  for (let i = 0; i < path.length; i += 1) {
+    const k = path[i];
+    if (yv instanceof Y.Map) {
+      pv = pv[k];
+      yv = yv.get(k as string);
+    } else if (yv instanceof Y.Array) {
+      const index = Number(k);
+      pv = pv[k];
+      yv = yv.get(index);
+    } else {
+      pv = null;
+      yv = null;
+    }
+  }
+
+  return { p: pv, y: yv };
+};
+
 export function bind<T>(
   p: Record<string, T> | T[],
   y: Y.Map<T> | Y.Array<T>,
   opts: Options = {},
-): void {
+): () => void {
```

Comment:
Would this result in a breaking change? It should probably be documented that the unsubscribe needs to be called.

Reply:
Breaking change in terms of the type, but not functionality. Everything can still be used as is, bind just returns a function to unsubscribe if desired.

Good idea, we could update the "how to use" section of the readme to show unbind capability.

Reply:
Sounds like a good improvement, because you can for example bind and unsubscribe in a useEffect to connect the proxy with a yDoc during a component life time.